gateway/api/views/jobs.py

```diff
@@ -249,6 +249,73 @@ def result(self, request, pk=None):  # pylint: disable=invalid-name,unused-argum
             serializer = self.get_serializer(job)
         return Response(serializer.data)
 
+    @action(methods=["POST"], detail=True)
+    def sub_status(self, request, pk=None):
+        """Update the sub status of a job."""
+        tracer = trace.get_tracer("gateway.tracer")
+        ctx = TraceContextTextMapPropagator().extract(carrier=request.headers)
+        with tracer.start_as_current_span("gateway.job.sub_status", context=ctx):
+            author = self.request.user
+            sub_status = self.request.data.get("sub_status")
+            if sub_status is None or sub_status not in Job.RUNNING_SUB_STATUSES:
+                return Response(
+                    {"message": "'sub_status' not provided or is not valid"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            def set_sub_status():
+                job = self.jobs_repository.get_job_by_id(pk)
+
+                if job is None:
+                    return Response(
+                        {"message": f"Job [{pk}] not found"},
+                        status=status.HTTP_404_NOT_FOUND,
+                    )
+
+                # If we import this with the regular imports,
+                # update jobs statuses test command fail.
+                # it should be something related with python import order.
+                from api.management.commands.update_jobs_statuses import (  # pylint: disable=import-outside-toplevel
+                    update_job_status,
+                )
+
+                update_job_status(job)
+
+                can_update_sub_status = JobAccessPolicies.can_update_sub_status(
+                    author, job
+                )
+                if not can_update_sub_status:
+                    return Response(
+                        {"message": f"Job [{job.id}] not found"},
+                        status=status.HTTP_404_NOT_FOUND,
+                    )
+
+                if job.status != Job.RUNNING:
+                    logger.warning(
+                        "'sub_status' cannot change because the job"
+                        " [%s] current status is not Running",
+                        job.id,
+                    )
+                    return Response(
+                        {
+                            "message": "Cannot update 'sub_status' when is not"
+                            f" in RUNNING status. (Currently {job.status})"
+                        },
+                        status=status.HTTP_403_FORBIDDEN,
+                    )
+
+                self.jobs_repository.update_job_sub_status(job, sub_status)
+
+                return Response({"message": "Sub status updated correctly"})
```

Comment:
Also, in a PATCH end-point, let's return the entity updated if everything was ok. In this case, let's return the Job with the `JobSerializerWithoutResult`, result would not be needed at this point.